Added initialize() and guardInitialized() to Process.

LightFuri opened the pull request:
Finished Error and Process tests.

LightFuri committed:
Added initialize() and guardInitialized() to Process. Finished Error and Process tests.

XAKEPEHOK reviewed (changes requested):
- Убрать guard в геттерах
- добавить isInitialized()

LightFuri committed:
Removed guardInitialized() from get-methods.
Added isInitialized() method.
jsonSerialize() now can return null in init and result if result is null or process have not been initialized.

LightFuri committed:
Removed guardInitialized() from terminate() and addError() methods. $init argument in initialize() now required and can be null.
Added test for null-init and non-init terminate().

XAKEPEHOK commented on src/Process.php:
```diff
@@ -161,23 +154,34 @@ public function __set($name, $value)
+                'value' => $this->init
+            ];
+        } else {
+            $init = null;
```
Это совершенно не критичный момент, но код можно сделать чуточку красивее. Здесь можно брать else и вынести `$init = null;` перед if

XAKEPEHOK commented on src/Process.php:
```diff
@@ -161,23 +154,34 @@ public function __set($name, $value)
+                'value' => $this->result,
+            ];
+        } else {
+            $result = null;
```
Аналогично

LightFuri committed:
Small code refactor

LightFuri committed:
Smallest code refactor
